### PR TITLE
Add basic git information

### PR DESCRIPTION
**Pull request by TessFerrandez**
This PR closes the following issues #271 #305 #306

- Add basic git information for cloning, branching committing, pushing, merging, resolving conflicts and handling remotes
- Add information on feature branching
- Separate the contributing information into separate files and remove the duplicated texts
- Update any existing links to files that moved

### CODE REVIEW

**Comment by TessFerrandez** [comment]
@annazietlow, @tessmichi - would love for you to review this if you have a chance

File: source-control/git-guidance/readme.md [hunk 1]
```diff
@@ -1,11 +1,198 @@
...
+git checkout -b feature/feature-name develop
+```
+
+At any point, you can move between the branches with `git checkout <branch>` as long as you have committed or stashed your work.
+If you forget the name of your branch use `git branch --all` to list all branches
+
+When you are done working, push your changes to a branch in the remote repository using `git push`
+
+```cmd
+git push --set-upstream origin feature/feature-name
```
Comment (annazietlow) [inline source-control/git-guidance/readme.md:70]: I would add an example of a regular 'git push' (once the branch is already linked to a remote branch) here too

**Review by annazietlow — APPROVED** [review]

File: source-control/contributing/merge-strategies.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+git fetch origin
+git checkout main
+git merge topic
+```
+
+## Two approaches to achieve a linear commit history
+
+### Rebase topic branch before merging into main
+
+Before merging `topic` into `main`, we rebase `topic` with the   :
```
Comment (JamesEarle) [inline source-control/contributing/merge-strategies.md:26]: "with the :" ??

File: source-control/git-guidance/readme.md [hunk 3]
```diff
@@ -1,11 +1,200 @@
...
+
+    ```cmd
+    git commit -m "add basic git instructions"
+    ```
+
+## Merging
+
+In [CSE](..\..\CSE.md) we encourage the use of Pull Request to merge code to the main repository to make sure that all code in the final product is [code reviewed](..\..\code-reviews\readme.md)
+
+The Pull Request (PR) process in [Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/repos/git/pull-requests?view=azure-devops), [GitHub](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request) and other similar tools make it easy both to start a PR, review a PR and merge a PR.
```
Comment (JamesEarle) [inline source-control/git-guidance/readme.md:116]: Might be nice to add a line about how you can link issues and ADO board items using #<number> for git issues or #AB<number> for ADO board items. This links them so when the PR is closed, they automatically move to closed.

**Review by JamesEarle — CHANGES_REQUESTED** [review]
Looks good, just very minor changes

**Review by fnocera — APPROVED** [review]
Looks great!

File: source-control/git-guidance/readme.md (hunk 3 above)
Comment (TessFerrandez) [reply source-control/git-guidance/readme.md:116]: This is already done in the code reviews section so I think I might leave it out of the git guidance

File: source-control/contributing/merge-strategies.md (hunk 2 above)
Comment (TessFerrandez) [reply source-control/contributing/merge-strategies.md]: Fixed

**Pull request merged**